src/Swarm/TUI/Controller.hs

```diff
@@ -99,64 +104,98 @@ handleEvent :: AppState -> BrickEvent Name AppEvent -> EventM Name (Next AppStat
 handleEvent s = case s ^. uiState . uiMenu of
   NoMenu -> handleMainEvent s
   MainMenu l -> handleMainMenuEvent l s
+  NewGameMenu l -> handleNewGameMenuEvent l s
   TutorialMenu -> pressAnyKey (MainMenu (mainMenu Tutorial)) s
-  ChallengesMenu -> pressAnyKey (MainMenu (mainMenu Challenges)) s
   AboutMenu -> pressAnyKey (MainMenu (mainMenu About)) s
 
 -- | The event handler for the main menu.
 handleMainMenuEvent ::
   BL.List Name MainMenuEntry -> AppState -> BrickEvent Name AppEvent -> EventM Name (Next AppState)
-handleMainMenuEvent l s (VtyEvent (V.EvKey V.KEnter [])) =
-  case snd <$> BL.listSelectedElement l of
-    Nothing -> continueWithoutRedraw s
-    Just x0 -> case x0 of
-      NewGame -> continue $ s & uiState . uiMenu .~ NoMenu
-      Tutorial -> continue $ s & uiState . uiMenu .~ TutorialMenu
-      Challenges -> continue $ s & uiState . uiMenu .~ ChallengesMenu
-      About -> continue $ s & uiState . uiMenu .~ AboutMenu
-      Quit -> halt s
-handleMainMenuEvent _ s (ControlKey 'q') = halt s
-handleMainMenuEvent menu s (VtyEvent ev) = do
-  menu' <- handleListEvent ev menu
-  continue $ s & uiState . uiMenu .~ MainMenu menu'
-handleMainMenuEvent _ s _ = continueWithoutRedraw s
+handleMainMenuEvent menu s = \case
+  VtyEvent (V.EvKey V.KEnter []) ->
+    case snd <$> BL.listSelectedElement menu of
+      Nothing -> continueWithoutRedraw s
+      Just x0 -> case x0 of
+        NewGame ->
+          continue $
+            s & uiState . uiMenu
+              .~ NewGameMenu (NE.fromList [mkScenarioList (s ^. gameState . scenarios)])
+        Tutorial -> continue $ s & uiState . uiMenu .~ TutorialMenu
```

Comment:
Well, my idea was as follows:

- Once we create "meta-scenarios" which can consist of a sequence of other scenarios, The Tutorial will be such a thing.
- We want the tutorial to be very prominently placed in the top-level menu, so someone just starting the game for the first time will immediately know how to start a tutorial.  Selecting that top-level Tutorial menu item will automatically load the meta-scenario consisting of the sequence of tutorial scenarios.
- However, it will still be possible to load individual tutorial scenarios through the New Game menu and selecting one manually.

Reply:
Sure, maybe we could leave a note that the tutorial is being slowly constructed over in the New game section?
```
In the meantime, you can play through a few tutorial parts over in the New game menu.
```